### ISSUE TITLE
Namespace compatibility break

### ISSUE THREAD

**Issue by chandon** [braintree/braintree_php#106]
The documentation should explain that the new version breaks the namespace compatibility.
To recover the compatibility, it's possible to use : 

    use Braintree\Configuration as Braintree_Configuration;
    use Braintree\ClientToken as Braintree_ClientToken;
    ...

**Comment by matthewpatterson** [comment]
@chandon This shouldn't be necessary since all classes are aliased to their old names (for example, [`Braintree_ClientToken`](https://github.com/braintree/braintree_php/blob/3.6.0/lib/Braintree/ClientToken.php#L49)).

We'll take a look and see what is going on here.

**Comment by crookedneighbor** [comment]
@chandon Would you provide a more complete code snippet so I can reproduce the error you're experiencing?

**Comment by crookedneighbor** [comment]
Are you using both the old names as well as the Braintree namespace? IE:

```php
namespace Braintree;

$result = Braintree_Transaction::sale([
    'amount' => '1000.00',
    'paymentMethodNonce' => 'fake-valid-nonce',
    'options' => [ 'submitForSettlement' => true ]
]);
```

This will not work as it'll interpret `Braintree_Transaction` as `Braintree\Braintree_Transaction`, which does not exist. You should instead do:

```php
$result = Braintree_Transaction::sale([
    'amount' => '1000.00',
    'paymentMethodNonce' => 'fake-valid-nonce',
    'options' => [ 'submitForSettlement' => true ]
]);
```

or

```php
namespace Braintree;

$result = Transaction::sale([
    'amount' => '1000.00',
    'paymentMethodNonce' => 'fake-valid-nonce',
    'options' => [ 'submitForSettlement' => true ]
]);
```

If you're experiencing something different, be sure to let us know.

**Comment by chandon** [comment]
In fact, i'm was using my own namespace, and so the use of braintree was declared like that : 

    namespace MyNameSpace;

    use Braintree_ClientToken;
    use Braintree_Configuration;
    use Braintree_Transaction;

    $result = Transaction::sale([
        'amount' => '1000.00',
        'paymentMethodNonce' => 'fake-valid-nonce',
        'options' => [ 'submitForSettlement' => true ]
    ]);

but now, i've to declare it like that because the old braintree's namespace seems to be not compatible anymore : 

    namespace MyNameSpace;

    use Braintree\ClientToken as Braintree_ClientToken;
    use Braintree\Configuration as Braintree_Configuration;
    use Braintree\Transaction as Braintree_Transaction;

    $result = Transaction::sale([
        'amount' => '1000.00',
        'paymentMethodNonce' => 'fake-valid-nonce',
        'options' => [ 'submitForSettlement' => true ]
    ]);

By the way, I think that : 

    namespace Braintree;

shouldn't be used in a project to declare the use of Braintree, it's not the purpose of "namespace", because all further declarations (like $result) we'll be done under "Braintree" namespace, even if it doesn't concern Braintree (the "braintree" namespace should'nt contain anything else than methods and properties declared in the braintree_php package)

**Comment by chandon** [comment]
@matthewpatterson Ok i think i got it. I'm using a frameword (Symfony). I think that Symfony's cache compiler check that the class "exists" before executing it. So, because the check appens before the inclusion, Symfony throw an error BEFORE including the file class containing "class_alias".

Perhaps all alias_class should be included outside the class file, in a separate file ? (i'm not sure about this solution)

**Comment by crookedneighbor** [comment]
Of course, I was just trying to reproduce an error with the data I had. :smile: 

Thanks for the information. We'll troubleshoot and try to reproduce the error with a Symfony integration and report back.

I was able to run your code snippet with a vanilla integration, so I think you're right that it has to do with Symfony's cache compiler.

**Comment by crookedneighbor** [comment]
@chandon We were able to reproduce the issue and patched it with [version 3.6.1](https://github.com/braintree/braintree_php/releases/tag/3.6.1). Thanks for bringing it to our attention!
